From: Hannes Reinecke <hare@suse.de>
Date: Wed, 18 Jul 2018 10:07:53 +0200
Subject: [PATCH] Revert "nvme: mark nvme_queue_scan static"
References: bsc#1054245
Patch-Mainline: never, kABI fix

That patch breaks the kABI, so we need to revert it.

Signed-off-by: Hannes Reinecke <hare@suse.com>
---
 drivers/nvme/host/core.c |    3 ++-
 drivers/nvme/host/nvme.h |    1 +
 2 files changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -117,7 +117,7 @@ static void nvme_set_queue_dying(struct
 	blk_mq_unquiesce_queue(ns->queue);
 }
 
-static void nvme_queue_scan(struct nvme_ctrl *ctrl)
+void nvme_queue_scan(struct nvme_ctrl *ctrl)
 {
 	/*
 	 * Only new queue scan work when admin and IO queues are both alive
@@ -125,6 +125,7 @@ static void nvme_queue_scan(struct nvme_
 	if (ctrl->state == NVME_CTRL_LIVE)
 		queue_work(nvme_wq, &ctrl->scan_work);
 }
+EXPORT_SYMBOL_GPL(nvme_queue_scan);
 
 int nvme_reset_ctrl(struct nvme_ctrl *ctrl)
 {
--- a/drivers/nvme/host/nvme.h
+++ b/drivers/nvme/host/nvme.h
@@ -395,6 +395,7 @@ void nvme_stop_ctrl(struct nvme_ctrl *ct
 void nvme_put_ctrl(struct nvme_ctrl *ctrl);
 int nvme_init_identify(struct nvme_ctrl *ctrl);
 
+void nvme_queue_scan(struct nvme_ctrl *ctrl);
 void nvme_remove_namespaces(struct nvme_ctrl *ctrl);
 
 int nvme_sec_submit(void *data, u16 spsp, u8 secp, void *buffer, size_t len,
